numpy/_core/src/umath/ufunc_object.c

```diff
@@ -1602,7 +1406,7 @@ execute_ufunc_loop(PyArrayMethod_Context *context, int masked,
     }
 
     /* The reset may copy the first buffer chunk, which could cause FPEs */
-    if (NpyIter_ResetBasePointers(iter, baseptrs, NULL) != NPY_SUCCEED) {
+    if (NpyIter_Reset(iter, NULL) != NPY_SUCCEED) {
```

Comment:
delay bufalloc, need to always reset, but basepointers was only for crazy array-prep support.